tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -4819,36 +4802,36 @@ Status FragmentMetadata::store_fragment_min_max_sum_null_count(
 
 Status FragmentMetadata::store_processed_conditions(
     const EncryptionKey& encryption_key, uint64_t* nbytes) {
-  Status st;
-  Buffer buff;
+  auto serialize_processed_conditions = [this](Serializer& serializer) {
+    // Store num conditions.
+    uint64_t num = processed_conditions_.size();
+    serializer.write<uint64_t>(num);
 
-  // Store num conditions.
-  uint64_t num = processed_conditions_.size();
-  st = buff.write(&num, sizeof(uint64_t));
-  if (!st.ok()) {
-    return LOG_STATUS(Status_FragmentMetadataError(
-        "Cannot serialize fragment metadata; Writing num processed conditions "
-        "failed"));
-  }
+    for (auto& processed_condition : processed_conditions_) {
+      uint64_t size = processed_condition.size();
+      serializer.write<uint64_t>(size);
 
-  for (auto& processed_condition : processed_conditions_) {
-    uint64_t size = processed_condition.size();
-    st = buff.write(&size, sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(Status_FragmentMetadataError(
-          "Cannot serialize fragment metadata; Writing processed condition "
-          "size failed"));
+      serializer.write(processed_condition.data(), processed_condition.size());
     }
+  };
+  SizeComputationSerializer size_computation_serializer;
+  serialize_processed_conditions(size_computation_serializer);
 
-    st = buff.write(processed_condition.data(), processed_condition.size());
-    if (!st.ok()) {
-      return LOG_STATUS(
-          Status_FragmentMetadataError("Cannot serialize fragment metadata; "
-                                       "Writing processed condition failed"));
-    }
+  Tile tile;
+  if (!tile.init_unfiltered(
```

Comment:
Since we are going to use this pattern repeatedly in many of these Buffer-related changesets, please implement a `Tile::from_generic(storage_size_t size)` and use it in place of these lines (note that all of the other arguments here are constant, [x-ref](https://github.com/TileDB-Inc/TileDB/pull/3407/files#diff-a31c8991a9335c2015e10b80bbaa7b7c7a0ec56513dad67f19ff5ddebb23fd04R1914)).

Reply:
done